### PR TITLE
feat: release workflow

### LINKED ISSUE #399: Automate the release process
🐞 **Problem**

This project follows a rolling release. That means we need to update the main and minor tags every time a new release is done. This translates into deleting these tags in the remote and re-tagging again. All this is manually and prone to error.

💡 **Idea**

Two ideas come to my mind:

- Use `tox.ini` and code the logic there for different platforms. Logic must be: read VERSION file -> delete necessary tags locally and remotely -> generate new tags -> upload new tags

- Create a `release.yml` workflow

### PR DESCRIPTION
**Pull request by RobPasMue**
Closes #399

Now, whenever you create a tag of type ``vX.Y.Z``, a rolling release is created for tags ``vX`` and ``vX.Y``

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Rolling release
+on:
+  push:
+    tags:
+      - "v*.*.*"
```
Comment (jorgepiloto) [inline .github/workflows/release.yml:5]:
```suggestion
  push:
    branches:
      - release/*
    tags:
      - "v*.*.*"
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just ensuring that this only triggers in `release/` branches.

File: .github/workflows/release.yml (hunk 1 above)
Comment (RobPasMue) [reply .github/workflows/release.yml:5]: Yep - makes sense

**Review by jorgepiloto — APPROVED** [review]
Awesome!

**Review by MaxJPRey — COMMENTED** [review]
Thanks for making our deployment more easy and reliable.

**Pull request merged**